src/python/WMCore/WMSpec/WMTask.py

```diff
@@ -926,10 +927,19 @@ def setSubscriptionInformation(self, custodialSites = None, nonCustodialSites =
             outputDataset = entry["outputDataset"]
             outputModule = entry["outputModule"]
 
-            primDs = outputDataset.split('/')[1]
-            tier = outputDataset.split('/')[3]
+            dsSplit = outputDataset.split('/')
+
+            primDs = dsSplit[1]
+
+            procDsSplit = dsSplit[2].split('-')
+            skim = ( len(procDsSplit) == 4 )
```

Comment:
You need two flags. I could have done the same with a trinary operator but as that doesn't exist in python, two booleans was the best choice (thought briefly about using True, False and None but that would have been just super ugly and dangerous).

4 is the number of '-' separated fields in the processed datasets name. The 4th is the filterName and is only used for skims. So if you have 4, the dataset in question is a "skim" by some definiiton. Good enough for the Tier0 at least, as all Tier0 output strictly follows processed dataset naming conventions.

I need two booleans because I need to be able to set subscription info

a) where it completely ignores whether the dataset is a skim or not (same as before)

b) differently based on whether the dataset is a skim or not